exercises/triangle/example.rs

```diff
@@ -0,0 +1,113 @@
+pub struct Triangle;
+
+impl Triangle {
+    #[cfg_attr(rustfmt, rustfmt_skip)]
+    fn valid_sides(sides: [u16; 3]) -> bool {
+        ( sides.iter().all(|&s| s > 0) )
+            && ( sides[0] + sides[1] >= sides[2] )
+            && ( sides[1] + sides[2] >= sides[0] )
+            && ( sides[2] + sides[0] >= sides[1] )
+    }
+}
+
+impl Triangle {
+    pub fn build(sides: [u16; 3]) -> Result<Box<TriangleType>, ()> {
+        if !Triangle::valid_sides(sides) {
+            Err(())
+        } else {
```

Comment:
in such cases I prefer to make the above a `return Err(())` and unindent everything in this `else` (removing the `else` completely)